iac-common/src/test/java/org/sonar/iac/common/yaml/TreePredicatesTest.java

```diff
@@ -30,11 +30,15 @@ class TreePredicatesTest {
   @Test
   void isTrue() {
     assertThat(TreePredicates.isTrue().test(text("true"))).isTrue();
+    assertThat(TreePredicates.isTrue().test(text("false"))).isFalse();
+    assertThat(TreePredicates.isTrue().test(text(null))).isFalse();
   }
 
   @Test
   void isEqualTo() {
     assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text("VALUE_TEST"))).isTrue();
+    assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text("NOT_VALUE_TEST"))).isFalse();
+    assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text(null))).isFalse();
```

Comment:
I was more thinking about another kind of tree which is not a ScalarTree aka TextTree